lib/graphs/digraph.nit

```diff
@@ -0,0 +1,561 @@
+# Implementations for representing directed graphs, also called digraphs.
+#
+# Currently, there are two classes
+module digraph
+
+# An arc of a digraph
+class Arc[V]
+	# The source of the arc
+	#
+	# ~~~
+	# var arc = new Arc[Int](0, 1)
+	# assert arc.source == 0
+	# ~~~
+	var source: V
+
+	# The target of the arc
+	#
+	# ~~~
+	# var arc = new Arc[Int](0, 1)
+	# assert arc.target == 1
+	# ~~~
+	var target: V
+
+	redef fun to_s: String do return "({source.to_s}, {target.to_s})"
```

Comment:
Return type is not needed on not covariant methods.

If you chose to keep the nullable bound you should add a check before. Furthermore, I think that `to_s` is automatically applied on something that is not a string located in a superstring so you can change your implementation by:

~~~nit
redef fun to_s do return "({source}, {target})"
~~~